node/web/api/booksApiController.js

```diff
@@ -26,5 +25,82 @@ module.exports.factory = function (router, repo, exceptions) {
         });
     });
 
+    router.post('/api/product', function (req, res) {
+      var count = "";
+      var bookTitle = "";
+      repo.get(req.query.product, function (err, book) {
+            if (err) {
+                exceptions.throwException(err);
+                res.status(400).end();
+                return;
+            }
+            var email = "";
+            console.log("Book title: " + book.title);
+            if(!req.cookies.email)
```

Comment:
Because you are arbitrarily setting the default email to "Guest" and then storing books in a cart, it means that *all* unauthenticated users of your app will share a shopping cart. Not good. To avoid this, you can either store the user's guest cart in their browser, or assign a unique session identifier to each user, and use that for their email. The problem with doing the latter, is that you will need to keep track of these carts and purge them at a point you decide the cart has been abandoned.